mmocr/datasets/preparers/data_obtainer.py

```diff
@@ -148,6 +149,14 @@ def move(self, mapping: List[Tuple[str, str]]) -> None:
         for src, dst in mapping:
             src = osp.join(self.data_root, src)
             dst = osp.join(self.data_root, dst)
+
+            if '*' in src.split('/')[-1] and not osp.exists(dst):
+                os.makedirs(dst)
+                for f in glob.glob(src):
+                    if osp.exists(f):
+                        shutil.move(f, dst)
+                continue
+
             if osp.exists(src) and not osp.exists(dst):
```

Comment:
These lines can be optimized to be more general. The `if` condition is too strict and may only be applicable to this dataset. 

1. Wildcard (*) does not usually appear in the file path, and therefore we can use `glob` to match all the files whenever there is `*` in the file path.
2. `mmengine` has a util `mkdir_or_exist` to create the directory whenever needed.
3. Is line 156 really necessary? I think `glob.glob` should return a list of existing files.
4. Using continue to skip the next `if` block doesn't make the logic flow very clear. I'm in favor of `if...elif...` here

To conclude, the code can be rewritten as:
```python
import mmengine

#...

if '*' in src:
    mmengine.mkdir_or_exist(dst)
    for f in glob.glob(src):
        shutil.move(f, dst)
elif osp.exists(src) and not osp.exists(dst):
```

Reply:
Thanks for your suggestions and I have modified this code block. But with https://github.com/open-mmlab/mmocr/pull/1639#discussion_r1056922636 fixed, this reports error like 'path already exists' when running task-recog after task-det. Keeping the `not osp.exists(dst) `in `if` statement will be fine. So it's like this 
```
if '*' in src and not osp.exists(dst):
    mkdir_or_exist(dst)
    for f in glob.glob(src):
        shutil.move(f, dst)

elif osp.exists(src) and not osp.exists(dst):
```

Reply:
It doesn't seem to address the key issue. Such an error occurs because `shutil` will check the existence of the destination file. If `dst` is just a directory path, `shutil` will do the check on the resulting filename, which is equivalent to `osp.exists(osp.join(dst, osp.basename(src))`.

Back to your implementation: `... and not osp.exists(dst)` only verifies the existence of the **directory** instead of a specific **file**. Suppose that we have placed an empty directory to `dst`, this implementation will still skip moving all the files inside. 

Therefore, it's better to just check the existence of each file with `osp.exists(osp.join(dst, osp.basename(f))` and only skip those existing ones.